app/src/main/java/com/xinto/opencord/ui/screens/pins/PinsScreen.kt

```diff
@@ -12,15 +12,18 @@ import androidx.compose.ui.input.nestedscroll.nestedScroll
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import com.xinto.opencord.R
+import com.xinto.opencord.ui.navigation.PinsScreenData
 import com.xinto.opencord.ui.viewmodel.ChannelPinsViewModel
 import kotlinx.collections.immutable.toImmutableList
 import org.koin.androidx.compose.getViewModel
+import org.koin.core.parameter.ParametersHolder
 
 @Composable
 fun PinsScreen(
+    data: PinsScreenData,
     onBackClick: () -> Unit,
     modifier: Modifier = Modifier,
-    viewModel: ChannelPinsViewModel = getViewModel()
+    viewModel: ChannelPinsViewModel = getViewModel(parameters = { ParametersHolder(mutableListOf(data)) })
```

Comment:
you can most likely do
```suggestion
    viewModel: ChannelPinsViewModel = getViewModel { parametersOf(data) }
```